mesonbuild/compilers/compilers.py

```diff
@@ -1610,18 +1625,44 @@ def __init__(self, compiler_type):
                           '2': default_warn_args,
                           '3': default_warn_args}
 
-        def get_module_incdir_args(self):
-            return ('-module', )
+    def get_module_incdir_args(self):
+        return ('-module', )
+
+    def get_no_warn_args(self):
+        return ['-silent']
 
-        def get_no_warn_args(self):
-            return ['-silent']
+    def openmp_flags(self):
+        return ['-mp']
+
+    def get_buildtype_args(self, buildtype: str) -> List[str]:
+        return pgi_buildtype_args[buildtype]
 
-        def openmp_flags(self):
-            return ['-mp']
+    def get_buildtype_linker_args(self, buildtype: str) -> List[str]:
+        return pgi_buildtype_linker_args[buildtype]
+
+    def get_optimization_args(self, optimization_level: str) -> List[str]:
+        return clike_optimization_args[optimization_level]
+
+    def get_debug_args(self, is_debug: bool) -> List[str]:
+        return clike_debug_args[is_debug]
+
+    def compute_parameters_with_absolute_paths(self, parameter_list: List[str], build_dir: str):
+        for idx, i in enumerate(parameter_list):
+            if i[:2] == '-I' or i[:2] == '-L':
+                parameter_list[idx] = i[:2] + os.path.normpath(os.path.join(build_dir, i[2:]))
 
     def get_allow_undefined_link_args(self):
         return []
 
+    def get_dependency_gen_args(self, outtarget, outfile):
+        return []
+
+    def get_always_args(self):
+        """PGI doesn't have -pipe."""
+        val = super().get_always_args()
+        val.remove('-pipe')
```

Comment:
This should not be implemented like this. Always args should be defined exactly for each compiler. Calling super here is not right because this class does not directly inherit from anything so this means it can only be used in a multiple inheritance context. In this case `super` can point to pretty much anything. Please return an explicit array of args instead (or the empty array if there are no always args).

Reply:
Oops a definite oversight, thanks